flink-metrics/flink-metrics-statsd/src/test/java/org/apache/flink/metrics/statsd/StatsDReporterTest.java

```diff
@@ -60,6 +60,20 @@
 public class StatsDReporterTest extends TestLogger {
 
 	@Test
+	public void testNameTruncating() {
+		StatsDReporter reporter = new StatsDReporter();
+
+		MetricConfig config = new MetricConfig();
+		config.setProperty(StatsDReporter.ARG_HOST, "localhost");
+		config.setProperty(StatsDReporter.ARG_PORT, "12345");
+		config.setProperty(StatsDReporter.ARG_MAX_COMPONENT_LENGTH, "10");
+		
+		reporter.open(config);
+		
+		assertEquals("0123456789", reporter.filterCharacters("0123456789DEADBEEF"));
```

Comment:
hmm, seems something was lost during the transfer :(

iirc, I wanted to ask whether it makes sense to also test that things like `a.b.0123456789DEADBEEF.c` are properly truncated to `a.b.0123456789.c` for the whole metric name